compiler/wasm/test/node/index.test.ts

```diff
@@ -36,32 +40,25 @@ describe('noir wasm compilation', () => {
   });
 
   describe('can compile scripts with dependencies', () => {
+    const sourceMap: PathToFileSourceMap = new PathToFileSourceMap();
     beforeEach(() => {
-      // this test requires a custom resolver in order to correctly resolve dependencies
-      initializeResolver((file) => {
-        switch (file) {
-          case '/script/main.nr':
-            return readFileSync(join(__dirname, depsScriptSourcePath), 'utf-8');
-
-          case '/lib_a/lib.nr':
-            return readFileSync(join(__dirname, libASourcePath), 'utf-8');
-
-          case '/lib_b/lib.nr':
-            return readFileSync(join(__dirname, libBSourcePath), 'utf-8');
-
-          default:
-            return '';
-        }
-      });
+      sourceMap.add_source_code('script/main.nr', readFileSync(join(__dirname, depsScriptSourcePath), 'utf-8'));
+      sourceMap.add_source_code('lib_a/lib.nr', readFileSync(join(__dirname, libASourcePath), 'utf-8'));
+      sourceMap.add_source_code('lib_b/lib.nr', readFileSync(join(__dirname, libBSourcePath), 'utf-8'));
     });
 
     it('matching nargos compilation', async () => {
-      const wasmCircuit = await compile('/script/main.nr', false, {
-        root_dependencies: ['lib_a'],
-        library_dependencies: {
-          lib_a: ['lib_b'],
+      const wasmCircuit = await compile(
+        'script/main.nr',
+        false,
+        {
+          root_dependencies: ['lib_a'],
+          library_dependencies: {
+            lib_a: ['lib_b'],
+          },
```

Comment:
Hmm, would this work with `lib_a/src/lib.nr`?

@kevaundray I forgot to mention it in our call yesterday but Aztec's noir-compiler wrapper uses source-resolver to dynamically remap library files to their location on disk. E.g. compiler would ask for `lib_a/lib.nr` and we'd use source-resolver to remap that to `/some/directory/lib_a/src/lib.nr`.

I think thi

Reply:
I think we'll have to remap the files when filling the `PathToFileSourceMap` so that the compiler finds the files where it expects them to be (e.g. map the file on disk `/home/alex/some_lib/src/lib.nr` to `some_lib/lib.nr`)

https://github.com/noir-lang/noir/blob/d120852473233488ba11aeb40f010b06f6cae735/compiler/wasm/src/compile.rs#L281-L284

Reply:
I think that we should have the dependency name and path be separated and both passed in tbh.

Currently we don't have a good way to handle a package which is depended on by two other packages which use different aliases for it, this will remove the need for remapping the files at the same time.

Reply:
Hmm, it should probably also handle aliasing libraries in case there's two crates depending on the same library (with the same name) but different versions. :thinking:

Reply:
I don't think these changes create any new issues for noir-compiler though right? You'll just need to move the path modification to be removing the root directory when adding it to the `PathToFileSourceMap` rather than removing it in the source-resolver.

We can fix this so that the wasm compiler doesn't have this strange behaviour later though I think.